src/main/java/mate/academy/bookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,130 @@
+import mate.academy.bookstore.service.ShoppingCartService;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final ShoppingCartService shoppingCartService;
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository itemRepository;
+    private final OrderItemMapper itemMapper;
+    private final OrderMapper orderMapper;
+
+    @Transactional
+    @Override
+    public void createOrder(OrderRequestDto orderDto, User user) {
+        ShoppingCart cart = shoppingCartService.getShoppingCart(user);
+        Set<CartItem> cartItems = cart.getCartItems();
+
+        Order order = createNewOrder(orderDto, user, cart);
+        Set<OrderItem> orderItems = createOrderItems(order, cartItems);
+        order.setOrderItems(orderItems);
+
+        orderRepository.save(order);
+    }
+
+    @Transactional
+    @Override
+    public List<OrderResponseDto> getAllOrders(User currentUser) {
+        return orderRepository.findByUserId(currentUser.getId()).stream()
+                              .map(orderMapper::toDto)
+                              .toList();
+    }
+
+    @Transactional
+    @Override
+    public void updateStatus(OrderStatusDto statusDto, Long orderId, Long userId) {
+        Order order = orderRepository.findById(orderId).orElseThrow(() ->
+                new EntityNotFoundException(
+                        "Unable to proceed: Order not found with id: " + orderId));
+        order.setStatus(statusDto.getStatus());
+        orderRepository.save(order);
+    }
+
+    @Override
+    public List<OrderItemResponseDto> getAllOrderItems(Long orderId) {
+        List<OrderItem> items = itemRepository.findByOrderId(orderId);
+
+        if (items.isEmpty()) {
+            throw new EntityNotFoundException(
+                    "Unable to proceed: No order items found for the order with id: " + orderId);
+        }
+        return items.stream()
+                    .map(itemMapper::toDto)
+                    .toList();
+    }
+
+    @Override
+    public OrderItemResponseDto getOrderItem(Long orderId, Long itemId, User user) {
+        Order order = orderRepository.findById(orderId).orElseThrow(() ->
+                new EntityNotFoundException(
+                        "Unable to proceed: Order not found with id: " + orderId));
+
+        if (!order.getUser().equals(user)) {
+            throw new AccessDeniedException(
+                    "User doesn't have permission to view order with id: " + orderId);
+        }
+        return itemRepository.findByIdAndOrderId(itemId, orderId)
+                             .map(itemMapper::toDto)
+                             .orElseThrow(() -> new EntityNotFoundException(
+                                     "Unable to proceed: Order item not found with id: " + itemId));
```

Comment:
Yoy can find order item by orderIt,, orderItemId and userId, and it will take only one db reuqest instead of 2 as implemented now